bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/protocol/philipstv/PhilipsTVConnectionManager.java

```diff
@@ -0,0 +1,697 @@
+        } catch (JsonProcessingException e) {
+            logger.warn("JsonProcessingException trying to save configMap: {}", e.getMessage(), e);
+        } catch (IOException ex) {
+            logger.debug("IOException when writing configJson to file {}", ex.getMessage());
+        }
+    }
+
+    private void readConfigs() {
+        String folderName = OpenHAB.getUserDataFolder() + "/androidtv";
+        String fileName = folderName + "/philipstv." + handler.getThing().getUID().getId() + ".config";
+        File file = new File(fileName);
+        if (!file.exists()) {
+            return;
+        }
+        try {
+            final byte[] contents = Files.readAllBytes(Paths.get(fileName));
+            String configJson = new String(contents);
+            logger.debug("Read configJson \"{}\" from {}", configJson, fileName);
+            Map<String, String> configMap = OBJECT_MAPPER.readValue(configJson,
+                    new TypeReference<HashMap<String, String>>() {
+                    });
+            this.username = Optional.ofNullable(configMap.get("username")).orElse("");
+            this.password = Optional.ofNullable(configMap.get("password")).orElse("");
+            this.macAddress = Optional.ofNullable(configMap.get("macAddress")).orElse("");
+            logger.debug("Processed configJson as {} {} {}", this.username, this.password, this.macAddress);
+        } catch (IOException ex) {
+            logger.debug("IOException when reading configJson from file {}", ex.getMessage());
+        }
+    }
+
+    public void setCreds(String username, String password) {
+        this.username = username;
+        this.password = password;
+        saveConfigs();
+    }
+
+    private boolean servicePing() {
+        int timeout = 500;
+
+        SocketAddress socketAddress = new InetSocketAddress(config.ipAddress, config.philipstvPort);
+        try (Socket socket = new Socket()) {
+            socket.connect(socketAddress, timeout);
+            return true;
+        } catch (ConnectException | SocketTimeoutException | NoRouteToHostException ignored) {
+            return false;
+        } catch (IOException ignored) {
+            // IOException is thrown by automatic close() of the socket.
+            // This should actually never return a value as we should return true above already
+            return true;
+        }
+    }
+
+    private void checkHealth() {
+        boolean isOnline = servicePing();
+        logger.debug("{} - Device Health - Online: {} - Logged In: {}", handler.getThingID(), isOnline, isLoggedIn);
+        if (isOnline != this.isOnline) {
+            this.isOnline = isOnline;
+            if (isOnline) {
+                logger.debug("{} - Device is back online.  Attempting reconnection.", handler.getThingID());
+                connect();
+            } else {
+                logger.debug("{} - Device is offline.", handler.getThingID());
```

Comment:
Remove logging when also setting Thing status.

Reply:
I have this in here for troubleshooting (hence the debug level).  It's helpful for me to see it all in one debug log from an end user rather than have to try to correlate it against the event log to see when the thing changed state.

Reply:
Fair enough.